cypress/e2e/timeline/timeline.spec.ts

```diff
@@ -305,6 +305,142 @@ describe("Timeline", () => {
+
+                // Check that the last EventTile is rendered
+                cy.get(".mx_EventTile.mx_EventTile_last").should("exist");
+            });
+
+            cy.get(".mx_MainSplit").percySnapshotElement("EventTiles on modern layout", { percyCSS });
+
+            // Check the same thing for compact layout
+            cy.setSettingValue("useCompactLayout", null, SettingLevel.DEVICE, true);
+
+            cy.get(".mx_MatrixChat_useCompactLayout").within(() => {
+                // Ensure CSS declarations which cannot be detected with a screenshot test are applied as expected
+                cy.get(".mx_EventTile")
+                    .should("have.css", "max-width", "100%")
+                    .should("have.css", "clear", "both")
+                    .should("have.css", "position", "relative");
+
+                // Check cascading works
+                cy.get(".mx_EventTile_continuation").should("have.css", "padding-block-start", "0px");
+
+                // Check that the last EventTile is rendered
+                cy.get(".mx_EventTile.mx_EventTile_last").should("exist");
+            });
+
+            cy.get(".mx_MainSplit").percySnapshotElement("EventTiles on compact modern layout", { percyCSS });
+        });
+
+        it("should render EventTiles on bubble layout", () => {
+            // Exclude timestamp and read marker from snapshots
+            const percyCSS = ".mx_MessageTimestamp, .mx_RoomView_myReadMarker { visibility: hidden !important; }";
+
+            sendEvent(roomId);
+            sendEvent(roomId); // check continuation
+            sendEvent(roomId); // check the last EventTile
+
+            cy.visit("/#/room/" + roomId);
+            cy.setSettingValue("layout", null, SettingLevel.DEVICE, Layout.Bubble);
+
+            // Wait until configuration is finished
+            cy.contains(
+                ".mx_RoomView_body .mx_GenericEventListSummary[data-layout=bubble] .mx_GenericEventListSummary_summary",
+                "created and configured the room.",
+            ).should("exist");
+
+            cy.get(".mx_RoomView_body[data-layout=bubble]").within(() => {
+                // Ensure CSS declarations which cannot be detected with a screenshot test are applied as expected
+                cy.get(".mx_EventTile")
+                    .should("have.css", "max-width", "none")
+                    .should("have.css", "clear", "both")
+                    .should("have.css", "position", "relative");
+
+                // Check that block start padding of the second message is not overridden
+                cy.get(".mx_EventTile.mx_EventTile_continuation").should("have.css", "margin-block-start", "2px");
+
+                // Select the last bubble
+                cy.get(".mx_EventTile_last")
+                    .within(() => {
+                        // calc(var(--gutterSize) - 1px)
+                        cy.get(".mx_EventTile_line").should("have.css", "padding-block-start", "10px");
```

Comment:
See: https://github.com/matrix-org/matrix-react-sdk/blob/2344eaab1c53b82b31721e5c8e36c723506734c4/res/css/views/rooms/_EventBubbleTile.pcss#L345-L349

This messy selector with the double negation should be simplified eventually.